source/UpcommingMovies.Infra.TheMovieDb.Test/MovieDiscoverServiceTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using UpcommingMovies.Infra.TheMovieDb.Services;
+using System.Threading.Tasks;
+using System.Globalization;
+using System.Threading;
+using System.Linq;
+
+namespace UpcommingMovies.Infra.TheMovieDb.Test
+{
+    [TestClass]
+    public class MovieDiscoverServiceTest
+    {
+        public MovieDiscoverServiceTest()
+        {
+            CultureInfo.DefaultThreadCurrentCulture = new CultureInfo("pt-BR");
+        }
+
+        private MovieDiscoverService _movieDiscoverService = new MovieDiscoverService(new TheMovieDbConfig()
+        {
+            ApiKey = "25ac607028f4f8915c576ead4be555df",
+            ApiBaseUri = new Uri("https://api.themoviedb.org/3/")
+        });
+
+        [TestMethod]
+        public async Task TestRetrieveUpCommingMovies()
+        {
+            var upcomming = await _movieDiscoverService.RetrieveMoviesAsync(1);
+            var x = upcomming.Movies.Select(m => m.BuildBackdropImageUri(500)).ToArray();
```

Comment:
I did this tests to support the movie db service client development. Since it depends of the movie db endpoint, it's not well suitable to unit test the app. I agree that is missing decoupled unit tests and its Asserts.